lib/HALSim/src/FileReader.cpp

```diff
@@ -0,0 +1,91 @@
+/* MIT License
+
+Copyright (c) 2023 Andreas Merkle <web@blue-andi.de>
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+
+*/
+
+/*******************************************************************************
+    DESCRIPTION
+*******************************************************************************/
+/**
+ *  @brief  File Reader
+ *  @author Gabryel Reyes <gabryelrdiaz@gmail.com>
+ */
+
+/******************************************************************************
+ * Includes
+ *****************************************************************************/
+
+#include "FileReader.h"
+#include <stdio.h>
+#include <Logging.h>
+
+/******************************************************************************
+ * Macros
+ *****************************************************************************/
+
+/******************************************************************************
+ * Prototypes
+ *****************************************************************************/
+
+/******************************************************************************
+ * Local Variables
+ *****************************************************************************/
+
+/******************************************************************************
+ * Public Methods
+ *****************************************************************************/
+
+bool FileReader::readFile(const String& fileName, char* outBuffer, const uint32_t maxBufferSize)
+{
+    bool  isSuccess = false;
+    FILE* file      = fopen(fileName.c_str(), "r");
+
+    if (nullptr == file)
+    {
+        LOG_ERROR("Error ocurred while opening configuration file \"%s\".", fileName.c_str());
+    }
+    else
+    {
+        size_t readBytes = fread(outBuffer, sizeof(char), maxBufferSize, file);

```

Comment:
What happens if the file is bigger than the maxBufferSize?

Reply:
Solved by checking EOF